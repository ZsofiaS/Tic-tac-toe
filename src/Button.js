import styled, { css } from 'styled-components';

const Button = styled.button`
  cursor: pointer;
  background: transparent;
  border-radius: 3px;
  color: palevioletred;
  border: 2px solid palevioletred;
  margin: 0 1em;
  padding: 0.25em 1em;
  transition: 0.5s all ease-out;

  // import css from styled components & interpolate a function into the template literal
  // the function gets passed the props of our component
  ${props =>
    props.primary &&
    css`
      color: crimson;
      border: 2px solid crimson;
    `};

  &:hover {
    background-color: ${(props) =>
      props.primary ? 'crimson' : 'palevioletred'};
    color: white;
  }
`;

export default Button;
