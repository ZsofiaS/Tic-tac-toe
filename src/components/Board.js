import React from 'react';
import styled from 'styled-components';

const BoardContainer = styled.div`
    width: 630px;
    height: 630px;
    padding: 1rem;
    border: 1px solid gainsboro;
    margin: 2rem auto;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
`;

const Square = styled.button`
    width: 200px;
    height: 200px;
    border: 1px solid crimson;
    margin: 4px;
`;

const Board = () => {
    return (
        <BoardContainer>
            <Square value="1" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="2" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="3" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="4" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="5" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="6" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="7" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="8" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="9" onClick={(e) => console.log(e.target.value)}></Square>
        </BoardContainer>
    )
}

export default Board;