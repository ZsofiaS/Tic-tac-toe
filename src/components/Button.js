import styled, { css } from 'styled-components';

const Button = styled.button`
  cursor: pointer;
  background: transparent;
  border-radius: 3px;
  color: black;
  border: 2px solid;
  border-image-slice: 1;
  border-image-source: linear-gradient(to left, #118ab2, #06d6a0, #ffd166);
  background-image: linear-gradient(to left, #118ab2, #06d6a0, #ffd166);
  margin: 0 1em;
  padding: 0.5em 0.8em;
  transition: 0.3s all ease-out;
  font-family: 'Grandstander', cursive;
  font-size: 18px;
  color: white;

  // import css from styled components & interpolate a function into the template literal
  // the function gets passed the props of our component
  ${(props) =>
    props.primary &&
    css`
      color: crimson;
      border: 2px solid crimson;
    `};

  &:hover {
    background-color: none;
    background-image: none;
    color: black;
  }
`;

export default Button;
