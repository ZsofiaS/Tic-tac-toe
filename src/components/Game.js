import React, { useState } from 'react';
import styled from 'styled-components';
import { Board } from './Board';

export const ResultTitle = styled.p`
  font-size: 40px;
  margin: 0.2rem;
`;
export const WinnerName = styled.span`
  background: -webkit-linear-gradient(319deg, #118ab2, #06d6a0, #ffd166);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  font-size: 55px;
`;

const Game = () => {
  const [currentPlayer, setCurrentPlayer] = useState('X');
  const [winner, setWinner] = useState('');

  const updatePlayer = () => {
    currentPlayer === 'X' ? setCurrentPlayer('O') : setCurrentPlayer('X');
  };

  const declareWinner = () => {
    currentPlayer === 'X' ? setWinner('O') : setWinner('X');
  };

  const declareTie = () => {
    setWinner('No one');
  };

  const restartGame = () => {
    setWinner('');
    setCurrentPlayer('X');
  };

  return (
    <>
      {winner !== '' ? (
        <ResultTitle>
          <WinnerName>{winner}</WinnerName> wins!
        </ResultTitle>
      ) : (
        <ResultTitle>
          <WinnerName>{currentPlayer}</WinnerName>'s turn
        </ResultTitle>
      )}
      <Board
        current={currentPlayer}
        updateHandler={updatePlayer}
        declareWinner={declareWinner}
        declareTie={declareTie}
        restart={restartGame}
      />
    </>
  );
};

export default Game;
