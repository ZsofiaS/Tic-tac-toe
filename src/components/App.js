import React from 'react';
import styled from 'styled-components';
import Game from './Game';
import GlobalStyle from '../globalStyles';

export const Container = styled.div`
  text-align: center;
  padding: 0.5rem;
`;

export const Title = styled.h1`
  font-size: 40px;
`;

export const Footer = styled.div`
  margin-top: 40px;
`;

function App() {
  return (
    <>
      <GlobalStyle />
      <Container>
        <Title>TIC-TAC-TOE</Title>
        <Game />
        <Footer>
          <p>Styled components & React Hooks</p>
          <p>
            Made with{' '}
            <span role="img" aria-label="Love">
              ðŸ’•
            </span>{' '}
            by Zsofi
          </p>
        </Footer>
      </Container>
    </>
  );
}

export default App;
