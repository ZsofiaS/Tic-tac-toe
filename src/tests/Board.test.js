import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Board, Square, BoardContainer } from '../components/Board';
import Button from '../components/Button';

let boardComponent;

beforeEach(() => {
  boardComponent = shallow(<Board />);
});

describe('renders Board component', () => {
  it('renders without crashing', () => {
    shallow(<Board />);
  });
  it('should render 9 squares', () => {
    expect(boardComponent.find(Square)).toHaveLength(9);
  });
  it('should render 1 boardContainer', () => {
    expect(boardComponent.find(BoardContainer)).toHaveLength(1);
  });
  it('should render 1 button', () => {
    expect(boardComponent.find(Button)).toHaveLength(1);
  });
  it('renders Board correctly', () => {
    expect(toJson(boardComponent)).toMatchSnapshot();
  });
});

describe('renders Square component', () => {
  it('renders without crashing', () => {
    shallow(<Square />);
  });
  it('should call onClick function when square clicked', () => {
    const mockedHandler = jest.fn();
    const clickHandler = mockedHandler;
    const square = shallow(<Square onClick={clickHandler} />);
    expect(mockedHandler).not.toHaveBeenCalled();
    square.simulate('click');
    expect(mockedHandler).toHaveBeenCalled();
  });
  it('renders Square correctly', () => {
    const squareComponent = shallow(<Square />);
    expect(toJson(squareComponent)).toMatchSnapshot();
  });
});
