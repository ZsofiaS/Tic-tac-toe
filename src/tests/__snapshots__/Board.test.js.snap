// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Board component renders Board correctly 1`] = `
<Fragment>
  <styled.div>
    <styled.button
      board={null}
      ended={false}
      id={0}
      key="0"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={1}
      key="1"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={2}
      key="2"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={3}
      key="3"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={4}
      key="4"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={5}
      key="5"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={6}
      key="6"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={7}
      key="7"
      onClick={[Function]}
    />
    <styled.button
      board={null}
      ended={false}
      id={8}
      key="8"
      onClick={[Function]}
    />
  </styled.div>
  <styled.button
    onClick={[Function]}
    type="button"
  >
    Start again
  </styled.button>
</Fragment>
`;

exports[`renders Square component renders Square correctly 1`] = `
<button
  className="sc-dlnjPT gdekQH"
/>
`;
